pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go

```diff
@@ -31,29 +34,30 @@ func (p plugin) MatchedPolicies(dataplane *core_mesh.DataplaneResource, resource
 	return matchers.MatchedPolicies(api.MeshTrafficPermissionType, dataplane, resources)
 }
 
+func (p plugin) EgressMatchedPolicies(es *core_mesh.ExternalServiceResource, resources xds_context.Resources) (core_xds.TypedMatchingPolicies, error) {
+	return matchers.EgressMatchedPolicies(api.MeshTrafficPermissionType, es, resources)
+}
+
 func (p plugin) Apply(rs *core_xds.ResourceSet, ctx xds_context.Context, proxy *core_xds.Proxy) error {
-	if proxy.Dataplane == nil {
-		// MeshTrafficPermission policy is applied only on DPP
-		// todo(lobkovilya): add support for ExternalService and ZoneEgress, https://github.com/kumahq/kuma/issues/5050
-		return nil
+	if proxy.ZoneEgressProxy != nil {
+		return p.configureEgress(rs, proxy)
 	}
 
-	if proxy.Dataplane.Spec.IsBuiltinGateway() {
-		return nil
-	}
-
-	mtp, ok := proxy.Policies.Dynamic[api.MeshTrafficPermissionType]
-	if !ok {
+	if proxy.Dataplane == nil || proxy.Dataplane.Spec.IsBuiltinGateway() {
 		return nil
 	}
 
 	if !ctx.Mesh.Resource.MTLSEnabled() {
 		log.V(1).Info("skip applying MeshTrafficPermission, MTLS is disabled",
-			"proxyName", proxy.Dataplane.GetMeta().GetName(),
```

Comment:
TBH I think it's fine to keep "proxyName". Otherwise, users can be confused why there are many absolutely identical messages. Another reason if you're filtering logs using proxyName you won't get this message